### ISSUE TITLE
Fix CreateUtcTimes migration

### ISSUE THREAD

**Issue by PhantomWatson** [BallStateCBER/muncie_events3#43]
Remove commented-out code from the CreateUtcTimes migration and move it to a RemoveTimeColumns migration, to be committed once nothing needs those columns anymore.

**Sidenote:** Despite earlier reckonings, all things considered, I think we should keep the date field as-is. It should make filtering by date way simpler than converting timezone-adjusted `Cake\I18n\Time` objects into and out of 'YYYY-MM-DD' strings.

**Comment by ericadeefox** [comment]
I don't understand... so, are we gonna keep the `date` field then, and have `start` & `end` times? I'm not sure what you mean.

**Comment by PhantomWatson** [comment]
Yeah, I suggest that it's better to leave the `date` column than to remove it.

I think we'll still need a second set of start/end time columns if we're migrating from storing them in Muncie time to UTC.

**Comment by ericadeefox** [comment]
OK cool. So that just means leave `date` alone? Don't convert it to UTC or anything like that? Because, like, if a `date` is `2018-01-01` for a party that starts 10pm Muncie time, then the UTC version of `date` would be for `2018-01-02`.

**Comment by PhantomWatson** [comment]
Here's my thinking. Here are the ways that stuff's going to be output:

To humans, who assume it's all in Muncie time:
 - December 15, 2017
 - 1:00pm

To API consumers (conforming to the `full-date` and `date-time` formats in [RFC 3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14)):
 - 2017-12-15
 - 2017-12-15T13:00:00-05:00

With times stored in UTC, when we tell CakePHP "output this time in RFC 3339 `date-time` format in our timezone", it should do that without any of the current weirdness.

I don't anticipate any problems with handling things this way, since we're always outputting everything in local time, just in a way that API consumers can interpret correctly. We could have the API output the date in `date-time` format, but then it would be identical to the "starting time" field, so they're not really gaining anything.

**Comment by ericadeefox** [comment]
Does this mean I should reverse a lot of the stuff I just wrote to treat the `start` values how `date` used to be treated? e.g. the datepicker going back to setting `date` col. Might be easier to just write a short method that sets `date` as the `date('Y-m-d')` version of `start`.

**Comment by PhantomWatson** [comment]
Unfortunately, yeah.

By "short method" do you mean making `date` a virtual field? I think we'd need `date` to remain an actual column to be able to do this:
 - Today is 2017-12-15
 - What events take place today?
 - Whatever events have `date = '2017-12-15'`

Instead of (I think) having to do this:
 - Today is 2017-12-15
 - What events take place today?
 - Whatever events have `start >= [2017-12-15T00:00:00+00:00 adjusted by timezone] AND start < [2017-12-16T00:00:00+00:00 adjusted by timezone]`

**Comment by ericadeefox** [comment]
I mostly mean in methods like add, edit, etc. right now I have it written to save `start` & `end`. So, if we're gonna continue to have `date` and keep saving `date`s too then, for the add & edit-type methods, I can just have it save `date` by taking `start` and changing it to Indiana time and setting it to `date('Y-m-d')` format.

and yeah going back to using `date` will simplify a lot of queries we have, even if it means undoing work I did. which, I mean, it's a pain in the ass but that's alright.

**Comment by ericadeefox** [comment]
Just to clarify: when viewing events, we want to get the date info from `$event->date` correct? Latest commit now saves `date` but still using the new-old system where it gets all its info from `start`.

**Comment by ericadeefox** [comment]
Actually, backpedal, I'm gonna leave all the stuff I did for the `start` field and I have a super super good idea why.

OK, so, let's say you have 3 events on a given date. Keeping them queried & ordered by the `start` field instead of the `date` field means they will be in order of time without need to write anything further. I still have it saving a `date` field, but I'm gonna leave all the queries I wrote that search for things by `start`.

**Comment by ericadeefox** [comment]
Oh, this is a fun, unexpected development that I should've totally expected. Removing the null datetime values means converting non-null `time_end` fields to `end`. Whoops. Time to account for that in the migrations.

**Comment by ericadeefox** [comment]
^^Those last three should be issue #48
